src/dbnode/storage/bootstrap/bootstrapper/readers.go

```diff
@@ -141,7 +141,9 @@ func enqueueReadersGroupedByBlockSize(
 				optimizedReadMetadataOnly, logger, span, nowFn, readInfoFilesResults)
 			readers[ShardID(shard)] = shardReaders
 		}
-		readersCh <- newTimeWindowReaders(group.Ranges, readers)
+		// Make sure to create before potentially being blocked on send.
```

Comment:
+1

Reply:
I think this is just so we have `newTimeWindowReaders(group.Ranges, readers)` precomputed, rather than waiting for it to execute on the send?

Reply:
It doesn't actually change the order of operations, it's still evaluated before pushing onto the channel.

I'll revert it, it was paranoia in case the runtime started doing something different in the future but AFAIK this is evaluated before anything is attempted to be sent on the channel. It's probably unlikely to change though in the runtime, I kind of mindlessly redid this.